consent-ws/src/main/java/org/broadinstitute/consent/http/service/ontology/IndexerUtils.java

```diff
@@ -1,35 +1,107 @@
+import org.semanticweb.owlapi.reasoner.structural.StructuralReasonerFactory;
+import org.semanticweb.owlapi.search.EntitySearcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import javax.ws.rs.InternalServerErrorException;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.util.*;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 /**
- * A suite of functions for handling index functions.
+ * A suite of functions for handling ontology indexing functions.
  */
 public class IndexerUtils {
 
-    private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(IndexerUtils.class);
+    private static final String FIELD_DEFINITION_PROPERTY = "IAO_0000115";
+    private static final String FIELD_HAS_EXACT_SYNONYM_PROPERTY = "hasExactSynonym";
+    private static final String FIELD_LABEL_PROPERTY = "label";
+    private static final String FIELD_DEPRECATED_PROPERTY = "deprecated";
+    private static final Logger logger = LoggerFactory.getLogger(IndexerUtils.class);
+    private static final ArrayList<String> IRI_FILTERS = new ArrayList<>(Arrays.asList("DOID", "DUOS", "DUO"));
 
     private Boolean isValidOWLClass(OWLClass owlClass) {
         return owlClass != null &&
             owlClass.isOWLClass() &&
             !owlClass.isOWLThing() &&
-            !owlClass.isOWLNothing();
+            !owlClass.isOWLNothing() &&
+            IRI_FILTERS.stream().anyMatch(f -> owlClass.getIRI().toString().contains(f));
+    }
+
+    void checkIndex(RestClient client, String indexName) {
+        try {
+            client.performRequest(
+                    "GET",
+                    ElasticSearchSupport.getIndexPath(indexName),
+                    ElasticSearchSupport.jsonHeader);
+        } catch (ResponseException re) {
+            // Response exception indicates a status code such as 404 not found, so try creating it.
+            try {
+                client.performRequest(
+                        "PUT",
+                        ElasticSearchSupport.getIndexPath(indexName),
+                        ElasticSearchSupport.jsonHeader);
+            } catch (IOException ioe) {
+                logger.error("Exception creating index: " + indexName + ": " + ioe.getMessage());
+            }
+        } catch (IOException e) {
+            logger.error("Exception checking for index: " + indexName + ": " + e.getMessage());
+        }
+    }
+
+    /**
+     * Generate terms from an ontology file, along with any direct ontology imports.
+     *
+     * @param streamRec The StreamRec that refers to an an ontology file.
+     * @return Collection of Terms generated from the ontology file
+     * @throws IOException The IOException
+     * @throws OWLOntologyCreationException The OWLOntologyCreationException
+     */
+    public Collection<Term> generateTerms(StreamRec streamRec) throws IOException, OWLOntologyCreationException {
```

Comment:
**Note for Reviewer:** This method is an extraction from where it was previously and is mostly unchanged, except to handle ontology imports.